### PR TITLE
docs: health report API, and diagnosis links (feature-targeted)

### PR DESCRIPTION
**Pull request by yaauie**
## Release notes

 [rn: skip]

## What does this PR do?

Documents the Health report API and provides `help_url` targets for the pipeline status probe that were delivered to the same feature branch in #16398 

## Why is it important/What is the impact to the user?

Docs & guidance are helpful :)

## Checklist

The same diff was previewed in https://github.com/elastic/logstash/pull/16513/files as a work-around for docs builds on PR's targeting feature branches getting skipped.

 - https://logstash_bk_16513.docs-preview.app.elstc.co/guide/en/logstash/8.16/logstash-health-report-api.html
 - https://logstash_bk_16513.docs-preview.app.elstc.co/guide/en/logstash/8.16/health-report-pipeline-status.html

### CODE REVIEW

File: docs/static/monitoring/monitoring-apis.asciidoc [hunk 1]
```diff
@@ -1184,3 +1184,155 @@ Example of a human-readable response:
...
+[[logstash-health-api-response-details]]
+==== Indicator Details
+
+Each health indicator in the health API returns a set of details that further explains the state of the system.
+The details have contents and a structure that is unique to each indicator.
+
+[[logstash-health-api-response-details-pipeline]]
+===== Pipeline Indicator Details
+
+`+pipelines/indicators/<pipeline_id>/details+`::
```
Comment (mashhurs) [inline docs/static/monitoring/monitoring-apis.asciidoc:1323]: It doesn't seem ASCIIDOC is catching up `+` here, preview has literal + sign.
```suggestion
`pipelines/indicators/<pipeline_id>/details`::
```

File: docs/static/monitoring/monitoring-apis.asciidoc [hunk 2]
```diff
@@ -1184,3 +1184,155 @@ Example of a human-readable response:
...
+(object) Information about the specified pipeline.
++
+.Properties of `+pipelines/indicators/<pipeline_id>/details+`
```
Comment (mashhurs) [inline docs/static/monitoring/monitoring-apis.asciidoc:1326]: Same literal + sign with above case.
```suggestion
.Properties of `pipelines/indicators/<pipeline_id>/details`
```

**Comment by yaauie** [comment]
Removing the plus-for-passthrough (via #16513):

Screenshot 2024-10-08 at 1 00 49 PM

**Review by mashhurs — APPROVED** [review]
🎉 Thank you!

**Pull request merged**